### DISCUSSION TITLE
FetchQuery fetch cached data refresh data in the background

### DISCUSSION THREAD

**Discussion by andredewaard** [TanStack/query#6989] [Q&A] [answered]
Is it possible with FetchQuery to fetch the cached data but still refresh the data in the background after?
Im fetching the authenticated user in the beforeEach route but want to refresh the user data if the staleTime is met but don't want to wait on that.

so im doing
```ts
router.beforeEach(async (route) => {
  const queryClient = inject('queryClient', new QueryClient())
  const data = await queryClient.fetchQuery({
    queryKey: ['auth', 'getMe'],
    queryFn: async () => {
      const data = await authService().getMe()
      return AuthUserModel(data.data)
    },
    staleTime: 1000 * 60 * 60 // 1 hour
    gcTime: 1000 * 60 * 60 * 5 // 5 hours
  })
  if (route.meta.requiresAuth && !data.id) {
    return { name: 'auth-login' }
  }
})
```
If its been an hour i want the data to be refreshed but in the background, the fetchQuery should still return the cached data so the user doesn't need to wait for the api to refresh before navigating to a different page.

**Comment by TkDodo** [comment, marked as answer]
we have a PR open for `ensureQueryData` to do that:

- https://github.com/TanStack/query/pull/6996

would that work for you?

**Comment by andredewaard** [comment]
That’s my PR! Sorry forgot about this issue.

**Reply by TkDodo** [reply]
Oh sorry I didn't check who's who 😂🙈
